wipac_dev_tools/setup_tools.py

```diff
@@ -0,0 +1,184 @@
+        py_min_max -- the min and max supported python releases: Ex: `((3,6), (3,8))`
+        description -- a one-line description of your package
+
+    Keyword arguments:
+        requirements_dir -- a sub-directory containing `requirements.txt`
+                            (if `requirements.txt` is in the root, ignore arg)
+        init_version_dir -- a sub-directory containing `__init__.py` which has `__version__` string
+                            (if this `__init__.py` is in the root, ignore arg)
+    """
+
+    def __init__(
+        self,
+        package_name: str,
+        abspath_to_root: str,
+        py_min_max: Tuple[Tuple[int, int], Tuple[int, int]],
+        description: str,
+        requirements_dir: str = "",
+        init_version_dir: str = "",
+    ):
+        self.name = package_name
+        self.here = abspath_to_root
+        self.py_min = min(py_min_max)
+        self.py_max = max(py_min_max)
+        self.requirements_path = os.path.join(
+            self.here, requirements_dir, "requirements.txt"
+        )
+        self.version = self._find_version(init_version_dir)
+
+        # dicts to be used like: `**this_kwarg`
+        self.author_kwargs = {
+            "author": "IceCube Collaboration",
+            "author_email": "developers@icecube.wisc.edu",
+        }
+        self.description_kwargs = {
+            "description": description,
+            # include new-lines
+            "long_description": open(os.path.join(self.here, "README.md")).read(),
+            "long_description_content_type": "text/markdown",
+            "keywords": description.split() + self.name.split("_"),
+        }
+
+        self._ensure_python_compatibility()
+
+    def _ensure_python_compatibility(self) -> None:
+        """If current python version is not compatible, warn and/or exit."""
+        if sys.version_info < self.py_min:
+            print(
+                f"ERROR: {self.name} requires at least Python "
+                f"{self.py_min[0]}.{self.py_min[1]} to run "
+                f"( {sys.version_info} < {self.py_min} )"
+            )
+            sys.exit(1)
+        elif sys.version_info > self.py_max:
+            print(
+                f"WARNING: {self.name} does not officially support Python "
+                f"{sys.version_info[0]}.{sys.version_info[1]}+ "
+                f"( {sys.version_info} > {self.py_max} )"
+            )
+
+    def _find_version(self, init_version_dir: str) -> str:
+        """Grab the package's version string."""
+        fpath = os.path.join(self.here, self.name, init_version_dir, "__init__.py")
+        with open(fpath) as init_f:
+            for line in init_f.readlines():
+                if "__version__" in line:
+                    # grab "X.Y.Z" from "__version__ = 'X.Y.Z'" (quote-style insensitive)
+                    return line.replace('"', "'").split("=")[-1].split("'")[1]
+
+        raise Exception(f"cannot find `__version__` string in '{fpath}'")
```

Comment:
I'll give it a shot

Reply:
Following up post-merge: this seemed to work, but failed once an outside `setup.py` tried to use `SetupShop`. If the `__init__.py` includes third-party imports, this fails because those packages have not yet been installed. This SO post goes into more detail: https://stackoverflow.com/a/2073599/13156561